kadi/commands/states.py

```diff
@@ -1522,20 +1534,34 @@ def set_transitions(cls, transitions, cmds, start, stop):
             elif tlmsid == "WSFEPALLUP":
                 transitions[date].update(fep_count=6, power_cmd=tlmsid)
 
-            elif tlmsid.startswith("WC"):
-                transitions[date].update(si_mode="CC_" + tlmsid[2:7])
-
             # Two special-case raw-mode SI modes
             # (https://github.com/sot/cmd_states/issues/23)
             elif tlmsid == "WT000B5024":
                 transitions[date].update(si_mode="TN_000B4")
 
             elif tlmsid == "WT000B7024":
                 transitions[date].update(si_mode="TN_000B6")
-
-            elif tlmsid.startswith("WT"):
-                transitions[date].update(si_mode="TE_" + tlmsid[2:7])
-
+            
+            # Special case for NIL SI modes 
+            elif tlsmid in NIL_SIMODES:
+            
+                translations[date].update(si_mode=NIL_SIMODES[tlmsid])
+                
+            # All other SI modes: this logic uses the PBLK command to 
+            # determine the SI mode hex string, which depends in part on
+            # whether a bias is being (re)computed.
+            elif tlmsid[:2] in ("WT", "WC"):
+                mode = {'WT': 'TE', 'WC': 'CC'}[tlmsid[:2]]
+                # find the hex digits in the PBLK command
+                digits = int(pblk_cmd[2:7], 16)
```

Comment:
Just in passing, what is pblk_cmd?

Reply:
Good catch, @jeanconn, that should have been `tlmsid`